File: DoItTest.Domain/Users/UserBlank.cs
using System.Text.Json.Serialization;

namespace DoItTest.Domain.Users
{
    public class UserBlank
    {
        public Guid? Id { get; set; }
        public String? Login { get; set; }
        public String? Password { get; set; }
        public UserRole? Role { get; set; }

        [JsonIgnore]
        public String PasswordHash => User.DefinePasswordHash(Password!);
    }
}

File: DoItTest/ClientApp/src/domain/users/userBlank.ts
import { User } from './user';
import { RoleType } from './userType';

export class UserBlank {
    public constructor(
        public id: string | null,
        public login: string | null,
        public password: string | null,
        public role: RoleType | null
    ) { }

    public static getDefault(): UserBlank {
        return new UserBlank(null, null, null, null)
    }

    public static formUser(user: User): UserBlank {
        return new UserBlank(user.id, user.login, user.password, user.role)
    }
}
